src/main/java/com/redhat/devtools/lsp4ij/LanguageServiceAccessor.java

```diff
@@ -177,21 +176,23 @@ public boolean hasAny(@NotNull VirtualFile file,
 
     @NotNull
     public CompletableFuture<List<LanguageServerItem>> getLanguageServers(@NotNull VirtualFile file,
-                                                                          @Nullable Predicate<ServerCapabilities> filter) {
-        return getLanguageServers(file, filter, null);
+                                                                          @Nullable Predicate<LSPClientFeatures> beforeStartingServerFilter,
+                                                                          @Nullable Predicate<LSPClientFeatures> afterStartingServerFilter) {
+        return getLanguageServers(file, beforeStartingServerFilter, afterStartingServerFilter, null);
     }
 
     @NotNull
-    public CompletableFuture<List<LanguageServerItem>> getLanguageServers(@NotNull VirtualFile file,
-                                                                          @Nullable Predicate<ServerCapabilities> filter,
-                                                                          @Nullable LanguageServerDefinition matchServerDefinition) {
+    private CompletableFuture<List<LanguageServerItem>> getLanguageServers(@NotNull VirtualFile file,
+                                                                           @Nullable Predicate<LSPClientFeatures> beforeStartingServerFilter,
+                                                                           @Nullable Predicate<LSPClientFeatures> afterStartingServerFilter,
+                                                                           @Nullable LanguageServerDefinition matchServerDefinition) {

```

Comment:
please don't break the API

Reply:
LanguageServiceAcessor have a comment which say that you must not use it.

I need really to break it,because the filter uses ServerCapabilties which is not enough to manage documentSelector.